scripts/buildscripts/install-build-pre-requisites.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+# Install following build pre-requisites for the Linux Agent
+# 1. install go lang and set the required environment variables
+# 2. install build dependencies
+# 3. install the docker engine
+set -e
+TEMP_DIR=temp-$RANDOM
+install_go_lang()
+{
+  echo "installing go 1.14.1 version ..."
+  echo "creating temp directory":$TEMP_DIR
+  sudo mkdir $TEMP_DIR && cd $TEMP_DIR
+  sudo curl -O https://dl.google.com/go/go1.14.1.linux-amd64.tar.gz
+  sudo tar -xvf go1.14.1.linux-amd64.tar.gz
+  sudo mv go /usr/local
+  echo "setting PATH and GOBIN environment variables"
+  export PATH=$PATH:/usr/local/go/bin
+  export GOBIN=/usr/local/go/bin
+  echo "export PATH=$PATH:/usr/local/go/bin" >> ~/.bashrc
+  echo "export GOBIN=/usr/local/go/bin" >> ~/.bashrc
+  source ~/.bashrc
+  echo "installation of go 1.14.1 and setting of PATH and GOBIN environment variables completed."
+}
+
+install_build_dependencies()
```

Comment:
will be great if we can have windows equivalent as well :)

Reply:
> will be great if we can have windows equivalent as well :)
> giving more work to me :) .. sure, let me try to get this.